### PR TITLE
ARROW-2133: [Python] Fix segfault on conversion of empty nested array to Pandas

### PR DESCRIPTION
**Pull request by pitrou**
Based on PR #1588

### CODE REVIEW

**Review by wesm — APPROVED** [review]
+1, thanks @pitrou! will await build (just rebased)

**Comment by pitrou** [comment]
There were apt problems in the C++ builds.

**Comment by wesm** [comment]
Here's the passing appveyor build, I just restarted the failed Linux builds: https://ci.appveyor.com/project/pitrou/arrow/build/1.0.65

File: cpp/src/arrow/python/arrow_to_pandas.cc [hunk 1]
```diff
@@ -519,7 +519,7 @@ inline Status ConvertListsLike(PandasOptions options, const std::shared_ptr<Colu
   // TODO(ARROW-489): Currently we don't have a Python reference for single columns.
   //    Storing a reference to the whole Array would be to expensive.
 
-  OwnedRef owned_numpy_array;
+  OwnedRefNoGIL owned_numpy_array;
```
Comment (xhochy) [inline cpp/src/arrow/python/arrow_to_pandas.cc]: Ah, that's nice!

**Review by xhochy — APPROVED** [review]

**Pull request closed without merging**